ext/bcmath/libbcmath/src/str2num.c

```diff
@@ -49,85 +48,96 @@ bool bc_str2num(bc_num *num, char *str, size_t scale)
 	/* Check for valid number and count digits. */
 	ptr = str;
 
-	if ((*ptr == '+') || (*ptr == '-')) {
-		/* Skip Sign */
-		ptr++;
-	}
+	ptr += (*ptr == '-' || *ptr == '+');
+
 	/* Skip leading zeros. */
 	while (*ptr == '0') {
 		ptr++;
 	}
-	/* digits before the decimal point */
+	integer_ptr = ptr;
+
+	/* Count digits. */
 	while (*ptr >= '0' && *ptr <= '9') {
 		ptr++;
 		digits++;
 	}
 	/* decimal point */
-	if (*ptr == '.') {
-		ptr++;
+	decimal_point = (*ptr == '.') ? ptr : NULL;
+
+	/* If a string other than numbers exists */
+	if (!decimal_point && *ptr != '\0') {
+		goto fail;
 	}
-	/* digits after the decimal point */
-	while (*ptr >= '0' && *ptr <= '9') {
-		if (*ptr == '0') {
-			trailing_zeros++;
-		} else {
-			trailing_zeros = 0;
+
+	/* search and validate fractional end if exists */
+	if (decimal_point) {
+		/* search */
+		fractional_ptr = decimal_point + 1;
+		fractional_end = fractional_ptr + strlen(fractional_ptr) - 1;
```

Comment:
See remark at the start of the function declaration

Reply:
ditto